app/src/main/java/org/kiwix/kiwixmobile/zimManager/ZimManageViewModel.kt

```diff
@@ -16,722 +16,723 @@
+       )
+     }
+     return None
+   }
+ 
+   private fun selectionsFromState() = fileSelectListStates.value?.selectedBooks.orEmpty()
+ 
+   private fun noSideEffectAndClearSelectionState(): SideEffect<Unit> {
+     fileSelectListStates.value?.let {
+       fileSelectListStates.postValue(
+         it.copy(
+           bookOnDiskListItems =
+             it.bookOnDiskListItems.map { booksOnDiskListItem ->
+               booksOnDiskListItem.apply { isSelected = false }
+             },
+           selectionMode = NORMAL
+         )
+       )
+     }
+     return None
+   }
+ 
+   @OptIn(ExperimentalCoroutinesApi::class)
+   private fun requestsAndConnectivityChangesToLibraryRequests(
+     library: MutableSharedFlow<LibraryNetworkEntity>,
+     dispatcher: CoroutineDispatcher = Dispatchers.IO
+   ) = requestDownloadLibrary.flatMapConcat {
+     connectivityBroadcastReceiver.networkStates.asFlow()
+       .distinctUntilChanged()
+       .filter { networkState -> networkState == CONNECTED }
+       .take(1)
+       .flatMapConcat {
+         shouldProceedWithDownload()
+           .flatMapConcat { kiwixService ->
+             downloadLibraryFlow(kiwixService).also {
+               onlineLibraryDownloading.tryEmit(true)
+             }
+           }
+       }
+   }
+     .filterNotNull()
+     .catch {
+       it.printStackTrace().also {
+         isOnlineLibraryDownloading = false
+         onlineLibraryDownloading.tryEmit(false)
+       }
+     }
+     .onEach {
+       library.emit(it).also {
+         // Setting this to true because once library downloaded we don't need to download again
+         // until user wants to refresh the online library.
+         isOnlineLibraryDownloading = true
+         onlineLibraryDownloading.tryEmit(false)
+       }
+     }
+     .flowOn(dispatcher)
+     .launchIn(viewModelScope)
+ 
+   private fun shouldProceedWithDownload(): Flow<KiwixService> {
+     return if (connectivityManager.isWifi()) {
+       flowOf(createKiwixServiceWithProgressListener())
+     } else {
+       flow {
+         val wifiOnly = sharedPreferenceUtil.prefWifiOnlys.first()
+         if (wifiOnly) {
+           shouldShowWifiOnlyDialog.postValue(true)
+           // Don't emit anything — just return
+           return@flow
+         }
+         emit(createKiwixServiceWithProgressListener())
+       }
+     }
+   }
+ 
+   private fun downloadLibraryFlow(
+     kiwixService: KiwixService
+   ): Flow<LibraryNetworkEntity?> = flow {
+     downloadProgress.postValue(context.getString(R.string.starting_downloading_remote_library))
+     val response = kiwixService.getLibrary()
+     downloadProgress.postValue(context.getString(R.string.parsing_remote_library))
+     emit(response)
+   }
+     .retry(5)
+     .catch { e ->
+       e.printStackTrace()
+       emit(LibraryNetworkEntity().apply { book = LinkedList() })
+     }
+ 
+   private fun updateNetworkStates() = connectivityBroadcastReceiver.networkStates
+     .asFlow()
```

Comment:
Here too.

As you have migrated into `stateFlow` so `catch` has no effect, and it is deprecated so please remove this as well.